Added Grafik Eye 3x/4x (via GRX-PRG/GRX-CI-PRG) to Lutron Binding

tmrobert8 opened the pull request:
Added Grafik Eye 3x/4x (via GRX-PRG/GRX-CI-PRG) to Lutron Binding

Signed-off-by: Tim Roberts troberts@bigfoot.com (github: tmrobert8)

kaikreuzer commented:
@actong May I ask you to review this and give your lgtm?

actong commented on addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/grxprg/SocketSession.java:
```diff
@@ -0,0 +1,385 @@
+                if (_isRunning.getAndSet(false)) {
+                    if (!_running.await(5, TimeUnit.SECONDS)) {
+                        _logger.warn("Waited too long for dispatcher to finish");
+                    }
```
Should this be closing the socket or something?  It's not clear to me how this will stop the thread unless some data comes down the wire and brings the thread out of the blocking read().

tmrobert8 commented:
1) Telnet - it's been so long but if I remember, there was some issue telnetclient and the prg communication that prevented it's usage.  I don't have have any other lutron products to test with and I didn't want to attempt modify your TelnetSession (for fear of screwing something up on your side).  
2) Line endings - do you mean I changed them on the whole project or just what I implemented?
3) Setting the "_isRunning" variable shuts down the socket because the thread.run will poll for one second and then check the variable (in the loop) and fall out of the loop - thus ending the thread.  The 5 second wait is there as a precaution (to not freeze the calling thread indefinately).

actong commented:
1. OK, was mostly curious on the rationale behind it.
2. On the files that were modified.  If you look at the commits in GitHub, it shows the whole file as having been changed, e.g. thing-types.xml.
3. I see the one second polling in Dispatcher, but not in ResponseReader, which looks like it'll get stuck in `c = _reader.read()` until some data comes down the wire.

tmrobert8 commented:
2) I'll check into but am curious why.  Editors are all set to unix newlines - not sure how that happened.
3) Ah - the setSoTimeout is set to 1 second when we create the client - so the read only blocks for a second - otherwise same logic

tmrobert8 commented:
Still have no idea how those files got that way - latest commit should fix those files

actong commented:
Gotcha.  Looks good to me.  Thanks for changing the line endings.

kaikreuzer commented:
@tmrobert8 Could you please also update the license headers and remove the "id" annotations in the JavaDoc here as well?

tmrobert8 commented:
Yes I can and done (too bad the header is obsolete this weeked [ie 2017])

kaikreuzer commented:
Many thanks, you are quick!
